### PR TITLE
add option to log momentum

### LINKED ISSUE #4354: Add momentum to the LearningRateMonitor
## 🚀 Feature

We should add a flag to optionally log the momentum value (for optimizers that have the attribute) in our LearningRateMonitor.

### Motivation

Some learning rate schedulers (eg. [OneCycle](https://pytorch.org/docs/stable/optim.html#torch.optim.lr_scheduler.OneCycleLR)) also adjust the momentum value when updating the learning rate. 

### Pitch

For each scheduler's optimizer, we can check to see if it has a momentum attribute. If so, we log the value in addition to the learning rate.

### PR DESCRIPTION
**Pull request by jeremyjordan**
## What does this PR do?

Adds the ability to optionally log momentum values in the LearningRateMonitor.

I followed the style of the existing code, except instead of saving all momentum values I decided to only save the last value. It seemed like this attribute was only being used for testing so I decided to keep the memory footprint lower.

Fixes #4354

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests? 
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review
 - [x] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.    
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/callbacks/lr_monitor.py [hunk 1]
```diff
@@ -123,9 +127,17 @@ def _extract_lr(self, trainer, interval: str) -> Dict[str, float]:
                         lr, key = pg['lr'], f'{name}/pg{i + 1}'
                         self.lrs[key].append(lr)
                         latest_stat[key] = lr
+                        if self.log_momentum and pg.get('momentum'):
+                            momentum_key, momentum_value = key + '-momentum', pg.get('momentum')
+                            self.last_momentum_values[momentum_key] = momentum_value
+                            latest_stat[momentum_key] = momentum_value
```
Comment (rohitgr7) [inline pytorch_lightning/callbacks/lr_monitor.py:133]: keep this in separate method? just to avoid repeated code.

Comment (jeremyjordan) [reply pytorch_lightning/callbacks/lr_monitor.py]: does the latest commit look better to you?

File: pytorch_lightning/callbacks/lr_monitor.py [hunk 2]
```diff
@@ -92,43 +95,62 @@ def on_train_start(self, trainer, *args, **kwargs):
...
                     for i, pg in enumerate(param_groups):
-                        lr, key = pg['lr'], f'{name}/pg{i + 1}'
-                        self.lrs[key].append(lr)
-                        latest_stat[key] = lr
+                        lr = self._extract_lr(param_group=pg, name=f'{name}/pg{i + 1}')
+                        latest_stat.update(lr)
+                        momentum = self._extract_momentum(param_group=pg, name=f'{name}-momentum/pg{i + 1}')
+                        latest_stat.update(momentum)
+
                 else:
```
Comment (s-rog) [inline pytorch_lightning/callbacks/lr_monitor.py:132]: Is this added to follow existing conventions in the code base? Otherwise seems a bit redundant?
(removing the idx suffix for single param groups)

File: pytorch_lightning/callbacks/lr_monitor.py (hunk 1 above)
Comment (rohitgr7) [reply pytorch_lightning/callbacks/lr_monitor.py]: yep.. better.

**Review by rohitgr7 — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

**Review by SkafteNicki — APPROVED** [review]

**Review by ydcjeff — APPROVED** [review]

File: pytorch_lightning/callbacks/lr_monitor.py (hunk 2 above)
Comment (jeremyjordan) [reply pytorch_lightning/callbacks/lr_monitor.py:132]: yes, i'm mirroring the existing convention for learning rates

**Review by s-rog — APPROVED** [review]
